src/org/opensolaris/opengrok/index/Indexer.java

```diff
@@ -101,542 +109,658 @@ private static void A_usage() {
+     * @throws ParseException 
+     */
+    public static String[] parseOptions(String[] argv) throws ParseException {
+        String[] usage = { "--help" };
+        String program = "opengrok.jar";
+        String[] onOff = {ON, OFF};
+        String[] remoteRepoChoices = {ON, OFF, DIRBASED, UIONLY};
 
-                // Let repository types to add items to ignoredNames.
-                // This changes env so is called after the setConfiguration()
-                // call above.
-                RepositoryFactory.initializeIgnoredNames(env);
+        if (argv.length == 0) {
+            argv = usage;  // will force usage output
+            status = 1;
+        }
 
-                if (noindex) {
-                    getInstance().sendToConfigHost(env, host, port);
-                    writeConfigToFile(env, configFilename);
-                    System.exit(0);
+        OptionParser configure = OptionParser.scan(parser -> {
+            parser.on("-R configPath").Do( cfgFile -> {
+                try {
+                cfg = Configuration.read(new File((String)cfgFile));
+                } catch(IOException e) {
+                    die(e.getMessage());
                 }
+            });
+        });
+
+        // An example of how to add a data type for option parsing
+        OptionParser.accept(WebAddress.class, s -> { return parseWebAddress(s); });
+
+        openGrok = OptionParser.Do(parser -> {
+
+            parser.setPrologue(
+                String.format("\nUsage: %s [options] [subDir1 [...]]\n", program));
+
+            parser.on("-?", "-h", "--help", "Display this usage.").Do( v -> {
+                parser.help();
+                System.exit(status);
+            });
+
+            parser.on(
+                "-A (.ext|prefix.):(-|analyzer)", "--analyzer", "/(\\.\\w+|\\w+\\.):(-|[a-zA-Z_0-9.]+)/",
+                    "Files with the named prefix/extension should be analyzed",
+                    "with the given analyzer, where 'analyzer' may be specified",
+                    "using a simple class name (CAnalyzer) or language name (C)",
+                    "(Note, analyzer specification is case sensitive)",
+                    "  Ex: -A .foo:CAnalyzer",
+                    "      will use the C analyzer for all files ending with .foo",
+                    "  Ex: -A bar.:C",
+                    "      will use the C analyzer for all files starting with bar.",
+                    "  Ex: -A .c:-",
+                    "      will disable the c-analyzer for for all files ending with .c").
+                Do( analyzerSpec -> {
+                    String[] arg = ((String)analyzerSpec).split(":");
+                    String fileSpec = arg[0];
+                    String analyzer = arg[1];
+                    configureFileAnalyzer(fileSpec, analyzer);
+                }
+            );
+
+            parser.on("-a", "--leadingWildCards", "=on/off", onOff, Boolean.class, 
```

Comment:
no short

Reply:
So are you asking to remove the -a option mentioned here and just use --leadingWildCards?

Reply:
yep